subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -140,12 +135,35 @@ static int transfer_request(struct coap_block_context *ctx,
 	}
 
 #if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_COAP_PROXY_SUPPORT)
+	/* if path is not available, off/len will be zero */
+	off = parsed_uri.field_data[UF_SCHEMA].off;
+	len = parsed_uri.field_data[UF_SCHEMA].len;
+	cursor = firmware_uri + off;
+
 	ret = coap_packet_append_option(&msg->cpkt, COAP_OPTION_PROXY_URI,
 					firmware_uri, strlen(firmware_uri));
 	if (ret < 0) {
 		SYS_LOG_ERR("Error adding PROXY_URI '%s'", firmware_uri);
 		goto cleanup;
 	}
+
+	/* TODO: convert to lower case */
+	if (strncmp(cursor, "http", 4) == 0) {
```

Comment:
This will match `https` as well. Is this OK?

Same for `coaps`, or anything that has `coap` or `http` as prefixes.

Reply:
I *think* this behavior is ok for now.  There could be some product which is able to handle DTLS and/or certificates on the proxy end and we probably shouldn't stop that from being an option.

If you want, I can add more validation here, but it seems like overkill, TBH as we would need to allow the user to specify the expected options.

Reply:
Both should be fine (https and coaps) as the proxy either validate the certificate from the server, or store the dtls key locally (as a proxy key).

Reply:
I don't think a little bit more validation here would be overkill; the schema is already being validated, but not completely.

If there's no problem in accepting a "httpfoobar" or "coap!@#)$()!@#($" scheme and adding that to the CoAP packet being built, then I'm fine with things the way they are in this patch.

Reply:
Good point.  Will update the schema checks.